### ISSUE TITLE
Magic

### ISSUE THREAD

**Issue by Hixie** [jakemac53/macro_prototype#15]
I really like how in the current setup, there's no magic. You can see exactly what code got generated. Debugging is easy since you walk through the generated code. Everything is nice and unambiguous.

As we work to streamline this I think we should try to maintain this transparency in some way.

**Comment by jakemac53** [comment]
Yes, that is a major advantage of building directly to the source tree. The final product almost certainly won't work in quite that same way, but we are in discussions about how to best enable the same functionality with the analyzer folks 👍 .
